torch/distributed/fsdp/_init_utils.py

```diff
@@ -1062,8 +1064,23 @@ def _sync_module_params_and_buffers(
         # Avoid re-synchronizing buffers in case of nested wrapping
         if not getattr(buffer, FSDP_SYNCED, False):
             setattr(buffer, FSDP_SYNCED, True)
-            module_states.append(buffer.detach())
-    module_states.extend(param.detach() for param in params)
+            detached_buffer = buffer.detach()
```

Comment:
nit: not sure if it is worth mentioning that this assumes no nested subclass (otherwise need to recursively call `__tensor_flatten__`)

Reply:
added a comment.